src/lib/ext-activitylog.js

```diff
@@ -64,8 +65,11 @@ class Model {
 class View {
   constructor() {
     this.logView = document.querySelector('log-view');
+    this.menuContainer = document.querySelector('.menu-container');
     this.clearLogBtn = document.querySelector('#clearLogBtn');
     this.saveLogBtn = document.querySelector('#saveLogBtn');
+    this.loadLogFile = document.querySelector('input[name="loadLogFile"]');
+    this.clearLoadedLogBtn = document.querySelector('#clearLoadedLogs');
```

Comment:
If the activitylog.html page for a loaded log file is only being used to review the logs in the loaded file, we may not need to different clear logs buttons, I have the feeling that it may just be confusing and the user may clear the logs being collected by mistake if it clicks the wrong button.

Reply:
@rpl Should we clear the loaded logs when a user clicks on clear logs buttons?

Reply:
If I recall correctly, when we discussed about this during the last meeting we agreed that clearing the loaded logs may not make sense.

Nevertheless, given that we are storing those logs in the background page, at some point we should be removing them, otherwise we may keep all of the loaded files around until Firefox is being restarted.

As a user, I would expect to "close the loaded file" more than "clear the loaded logs", but from an implementation perspective they are basically the same thing, just named differently to match what is more likely the user expectation.

And so (I'm basically thinking about this aloud) I feel that, when the activitylog page is this kind of "view logs from file" mode, it may be reasonable to:

- list the logs loaded from file (maybe in a select input element), and allow the user to switch between the loaded logs files
- have a button to "close the loaded logs file", which would remove the entry for that loaded logs from the background page

@wagnerand what do you think about this? would the behavior described above match your expectation as a user
(also @Rob--W for his opinion on this).

Reply:
I think expecting a user to have to click on _Close the loaded logs file_ might be a bit too much. It's not unlikely they forget to click the button and just close the tab, in which case the logs would not be removed from memory. I could think about 2 possible options to do this:

Option 1 (simple): When the tab gets closed, remove the logs from the background page. If the user decides to undo closing the tab, they have to reload the file.

Option 2 (a bit more involved): When the tab gets closed, start a timer (timeout tbd, but I guess 1 minute or so would do). When it times out, check if the log is loaded in a tab. If so, just do nothing. If not, remove the log from the background page.

Of course, the check whether a log is still shown somewhere, could be something else, for example a lock we set on the log data when the file gets loaded, and unset when the tab gets closed. After the timeout, we'd check whether the lock is set and remove the log from background page if it's not.